### PR TITLE
added windows specific stub

### PR DESCRIPTION
**Pull request by hoopoepg**
- some unix specific functions are moved into unix/osd.c
- added basic windows layer (to provide compilation in windows
  platform)

Signed-off-by: soblomov <sergey.oblomov@intel.com>

### CODE REVIEW

**Comment by lanl-ompi** [comment]
bot:retest

**Comment by lanl-ompi** [comment]
Test FAILed.

File: prov/sockets/include/sock.h [hunk 1]
```diff
@@ -1186,4 +1186,6 @@ int sock_epoll_wait(struct sock_epoll_set *set, int timeout);
 int sock_epoll_get_fd_at_index(struct sock_epoll_set *set, int index);
 void sock_epoll_close(struct sock_epoll_set *set);
 
+size_t sock_rx_avail_len(struct sock_rx_entry *rx_entry);
+
```
Comment (jithinjosepkl) [inline prov/sockets/include/sock.h]: We have this declaration already in sock.h

File: include/rdma/fabric.h [hunk 2]
```diff
@@ -481,6 +481,10 @@ struct fi_param {
 int fi_getparams(struct fi_param **params, int *count);
 void fi_freeparams(struct fi_param *params);
 
+#if defined(_WIN32)
+#include <BaseTsd.h>
+typedef SSIZE_T ssize_t;
+#endif
```
Comment (shefty) [inline include/rdma/fabric.h]: Please move this to the top of the file, so that future changes don't use ssize_t before it's defined on windows.

File: include/windows/osd.h [hunk 3]
```diff
@@ -0,0 +1,298 @@
...
+{
+	return -1;
+}
+
+#ifdef __cplusplus
+}
+#endif
+
+#endif /* _FI_WIN_OSD_H_ */
+
```
Comment (shefty) [inline include/windows/osd.h]: Can any of the definitions in this header file move into more appropriate headers?  For example, move the pthread stuff into windows/pthread.h, so the definitions are not split between two files.

File: libfabric.vcxproj.filters [hunk 4]
```diff
@@ -0,0 +1,332 @@
...
+    <ClInclude Include="prov\sockets\include\sock_util.h">
+      <Filter>Source Files\prov\sockets\include</Filter>
+    </ClInclude>
+  </ItemGroup>
+  <ItemGroup>
+    <None Include="libfabric.def">
+      <Filter>Source Files</Filter>
+    </None>
+  </ItemGroup>
+</Project>
```
Comment (shefty) [inline libfabric.vcxproj.filters]: Can you separate the project files into their own patch, so it's easier to review the source code changes?

File: prov/sockets/include/sock.h (hunk 1 above)
Comment (shefty) [inline prov/sockets/include/sock.h]: The function is modified below to change it from an inline.  Given the function, I would move the entire thing into the header file to force inlining.

File: include/windows/osd.h (hunk 3 above)
Comment (hoopoepg) [inline include/windows/osd.h]: done. specific headers are moved into appropriate files (similar to Linux disposition)

File: include/rdma/fabric.h (hunk 2 above)
Comment (hoopoepg) [inline include/rdma/fabric.h]: done

File: prov/sockets/include/sock.h (hunk 1 above)
Comment (hoopoepg) [inline prov/sockets/include/sock.h]: done: changed to inline function

File: include/rdma/fabric.h [hunk 5]
```diff
@@ -41,6 +41,10 @@
 extern "C" {
 #endif
 
+#if defined(_WIN32)
+#include <BaseTsd.h>
+	typedef SSIZE_T ssize_t;
+#endif
```
Comment (shefty) [inline include/rdma/fabric.h]: nit - I'd move this even higher, near the other include commands, outside of the extern "C"

Comment (hoopoepg) [inline include/rdma/fabric.h]: ok, upper :)

**Comment by shefty** [comment]
thanks!
